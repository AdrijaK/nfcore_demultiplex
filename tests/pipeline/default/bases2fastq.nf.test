nextflow_pipeline {

    name "Test Bases2Fastq"
    script "main.nf"

    test("Should run without failures") {

        when {
            params {
                input = "$baseDir/assets/inputs/b2fq-samplesheet.csv"
                demultiplexer = 'bases2fastq'
                outdir = "$outputDir"
            }
        }

        then {
            assert workflow.success
            assert workflow.trace.succeeded().size() == 8

            assert snapshot(
                // FIXME
                // path("$outputDir/sim-data/DefaultSample_R1.fastq.gz.md5"),
                // path("$outputDir/sim-data/DefaultSample_R2.fastq.gz.md5"),
                path("$outputDir/sim-data/Metrics.csv"),
                path("$outputDir/sim-data/RunManifest.json"),
                path("$outputDir/sim-data/UnassignedSequences.csv"),

                path("$outputDir/sim-data/Samples/DefaultSample/DefaultSample_R1.fastq.gz"),
                path("$outputDir/sim-data/Samples/DefaultSample/DefaultSample_R2.fastq.gz"),
            ).match()

            assert new File("$outputDir/sim-data/Bases2Fastq-Sim_QC.html").exists()
            assert new File("$outputDir/sim-data/RunStats.json").exists()
            assert new File("$outputDir/sim-data/Samples/DefaultSample/DefaultSample_stats.json").exists()
        }

    }

}
