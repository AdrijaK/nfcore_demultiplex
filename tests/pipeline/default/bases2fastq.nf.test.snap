{
    "Should run without failures": {
        "content": [
            "Metrics.csv:md5,3213ad7bdf66f33476cb589b4d608948",
            "RunManifest.json:md5,d4ecdb19e5fbcffab3e6e770c34023fb",
            "UnassignedSequences.csv:md5,11c1693830ce941b8cfb8d2431a59097",
            "DefaultSample_R1.fastq.gz:md5,77b68ec94323741d1bddc03a7a428fa9",
            "DefaultSample_R2.fastq.gz:md5,ecd4ed53e10207582bbbc29383080b57"
        ],
        "timestamp": "2023-01-18T23:53:51+0000"
    }
}